### DISCUSSION TITLE
Socket io connects multiple time when the server is hosted live

### DISCUSSION THREAD

**Discussion by saicharanreddythummala** [socketio/socket.io#4572] [Q&A] [answered]
Error
![image](https://user-images.githubusercontent.com/95035414/209575283-c3085b8b-900f-4b63-9112-4a9d924767aa.png)

Client
![image](https://user-images.githubusercontent.com/95035414/209575347-7b71f4b8-d35e-4134-8c60-1b33a4a4c15e.png)

Server
![image](https://user-images.githubusercontent.com/95035414/209575417-eeac53f1-3859-4ee4-953b-fa2eaab862d7.png)

I am not able to figure out the error. This only happens when the server is hosted on Vercel/Heroku. But when I run it locally there is no error. Kindly help me out!!!

Thanks

**Comment by qiulang** [comment]
your cros setting as `origin:'http://localhost:3000'` is wrong as you host your server on the cloud. 

Besides, if your client does not send cross-site requests, you don't need to set `withCredentials`

**Comment by darrachequesne** [comment, marked as answer]
Hi! The Vercel platform does not support WebSocket connections: https://vercel.com/guides/do-vercel-serverless-functions-support-websocket-connections

Please note that since most serverless platforms bill by the duration of the request handler, maintaining a long-running connection with Socket.IO (or plain WebSocket) is not recommended.

**Reply by saicharanreddythummala** [reply]
Thank you for the clarification. Much appreciated!!!
